src/layer.js

```diff
@@ -234,6 +234,34 @@ export class MapLayer extends HTMLElement {
       }
     }, 0);
   }
+  getOuterHTML() {
+    let fullLayer = this.outerHTML;
+    let tempElement = document.createElement('div');
+    tempElement.innerHTML = fullLayer;
+
+    if (this.hasAttribute('src')) {
+      let newSrc = this._layer.getHref();
+      tempElement.querySelector('layer-').setAttribute('src',newSrc);
+    }
+    if (this.querySelector('map-link')) {
+      let mapLinks = tempElement.querySelectorAll('map-link');
+
+      mapLinks.forEach((mapLink) => {
+       
+        if (mapLink.hasAttribute('href')) {
+          mapLink.setAttribute('href', decodeURI(this.transformURL(mapLink.attributes.href.value, this.baseURI ? this.baseURI : document.baseURI)));
+        }
+        else if (mapLink.hasAttribute('tref')) {
+          mapLink.setAttribute('tref', decodeURI(this.transformURL(mapLink.attributes.tref.value, this.baseURI ? this.baseURI : document.baseURI)));
+        }
+      });
+    }
+
+    return tempElement.innerHTML;
+  }
+  transformURL(url, baseUrl) {
```

Comment:
This member function doesn't seem necessary, tbh, it's just wrapping `new URL(url, base)).href` ; anyway the scope is public which is too much.  The scopy of it could even be **within** `getOuterHTML`, since it's defined for use by that function alone.  But, I would personally not define a function here, and more simply use `new URL(...).href`